tsdb/record/record.go

```diff
@@ -37,6 +37,8 @@ const (
 	Samples Type = 2
 	// Tombstones is used to match WAL records of type Tombstones.
 	Tombstones Type = 3
+	// Exemplars is used to match WAL records of type Exemplars.
+	Exemplars Type = 5
```

Comment:
4 was originally going to be metadata records from https://github.com/prometheus/prometheus/pull/7771 but since that's stale I'll use 4 for exemplars and that PR can be updated to use 5 for metadata.